flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/TimerSerializer.java

```diff
@@ -230,16 +264,29 @@ public TypeSerializerConfigSnapshot snapshotConfiguration() {
 	 */
 	public static class TimerSerializerConfigSnapshot<K, N> extends CompositeTypeSerializerConfigSnapshot {
 
+		private static final int VERSION = 0;
+
 		public TimerSerializerConfigSnapshot() {
 		}
 
-		public TimerSerializerConfigSnapshot(TypeSerializer<K> keySerializer, TypeSerializer<N> namespaceSerializer) {
-			super(keySerializer, namespaceSerializer);
+		public TimerSerializerConfigSnapshot(
+			@Nonnull TypeSerializer<K> keySerializer,
+			@Nonnull TypeSerializer<N> namespaceSerializer) {
+			super(init(keySerializer, namespaceSerializer));
```

Comment:
Just curious:
does this `init` method exist just to ensure the ordering between the `keySerializer` / `namespaceSerializer`?

Reply:
Yes, it is a bit more solid like this.